### DISCUSSION TITLE
0.10 + (Quark 0.5) - Back again

### DISCUSSION THREAD

**Discussion by XorTroll** [XorTroll/Goldleaf#633] [General]
# Changelog

In general, a huge code cleanup went through this project

## General

- The USB communication protocol between Goldleaf and Quark went through some relevant and breaking changes

## Goldleaf

- Updated for latest libusbhsfs/Plutonium changes, this last one probably means some UI bugs may have been fixed

- When browsing a webpage, the selected user is used

- Some improvements on the text/hex reader, which is now slightly faster

- New supported languages: Japanese, Portuguese, Brazilian, Simplified Chinese, Korean

- Global and user title playstats are shown instead of just user ones (when a user is selected)

- Selected images on the file browser are now shown on the options dialog, as a preview

- Support added for some recent installable contents (known as "sparse" contents)

- Other various bug fixes

## Quark

Quark has been fully rewritten:

- Removed Material style, since that feature (its libs) were causing the version issues with this project

- A single build is produced, which works for both Java 8/9+

- The window is now stretchable

- Removed Discord presence support since it wasn't such a relevant feature and due to conflicts with the libs which were used for it

<hr /><em>This discussion was created from the release <a href='https://github.com/XorTroll/Goldleaf/releases/tag/0.10'>0.10 + (Quark 0.5) - Back again</a>.</em>

**Comment by simonarhaegar** [comment]
Hi! I cant open quark, even tho I dowloaded Java (even from Explorer, since Chrome was said to have a problem). My computer says that my quark  is somehow compressed but when I unzip it it creates a giant folder with lots of files and no apllication to run. Wha should I do? I use windows.

**Reply by XorTroll** [reply]
Run it from the console with `java -jar Quark.jar` (on the directory where you downloaded/placed it)

**Comment by luc1dni9htmare** [comment]
Can you add a feature with Always ON? 
Prevents sleep on Nintendo Switch and Quark being disconnected. - The reconnect button/option terminates the application.

**Comment by Alaet** [comment]
Hi there, not sure about what i am about to say but after updating GoldLeaf, ns-usbloader(5.2) and GoldLeaf 0.10 crashes payload when trying to explore content through usb cable (not sure if i should rather say this to ns usbloader dev's but ... here i am, in case you know eachother or w/e) 
Anyway, i've went back to 0.9 but i'm sure you did a great job :) Cheers !

**Reply by XorTroll** [reply]
Is ns-usbloader updated for this new version?

**Reply by Alaet** [reply]
"Of course" not, but since i've noticed it, i figure out i should say it here in case other people wonder

**Comment by lookye** [comment]
The developer of ns-usbloader should be informed if its not compatible with the latest goldleaf release.

**Comment by Drarkusss0** [comment]
When I try to open Quark I got Unable to Connect via USB even tho it's connected and have Goldleaf v0.10 open

**Comment by bitb0xu** [comment]
man the language in brazil is not "brazilian", the language is portuguese

**Reply by Masamune3210** [reply]
To be fair, its Brazillian Portuguese, and to people who don't know any better, its a 50/50 shot of which side of the split they decide to shorten it to

**Comment by RoyalTaz** [comment]
Hello.
Is it possible, for the next version, to add the "install to SD card" option, when we install multi .NSP ?

**Comment by Jhonn250** [comment]
Hi, anyone having trouble opening the quark jar on macOS ventura?

**Comment by kleon7** [comment]
Hi, Goldleaf 0.10 seems to be not compatible with firmware 16, everytime I try to install a NSP I get this error code: 

Error 2356-0009
Module goldleaf (356)
PFS0 (NSP) not valid

What should I do to fix that?
Thanks

**Reply by stefmaster2411** [reply]
I get the same error

At first: 

Error: 2001-0123 (0xF601) 

and then, if I try again

Error 2356-0009

**Reply by Symbiote81062** [reply]
Same here. I followed everything correctly, but then when I go to install it gives 
Error 2001-0123 (0xF601)
Module: unknown module(1)
Decription: Unknown or undocumented error (123)

then after trying to install it again it gives 
Error 2356-0009
Module goldleaf (356)
Invalid PFS0 (NSP) (9)

**Reply by stefmaster2411** [reply]
At the moment i solved the issue installing Tinfoil, an alternative to Goldleaf
It work 100%

**Comment by Sgbdk** [comment]
hi all i need help.. i installed Goldleaf 0.10 to my switch but its show the atmosphere folder as a file and as a result im unable to access whatever is inside  the Atmosphere folder. Can someone please advice ?
![20230406_211945](https://user-images.githubusercontent.com/130059940/230393555-d085d10a-cc63-457b-a272-bb8bde686e1f.jpg)

**Comment by Ammako** [comment]
> Support added for some recent installable contents (known as "sparse" contents)

Might as well add support for NSZ and XCI now. Homebrew NSPs aren't going to have sparse NCAs, the only reason to add support for it is piracy which was the reason for not wanting to add support for NSZ/XCI.

[#417](https://github.com/XorTroll/Goldleaf/issues/417#issuecomment-539114241)
